dev/com.ibm.ws.javaee.ddmodel/src.gen/com/ibm/ws/javaee/ddmodel/webext/WebExtAdapter.java

```diff
@@ -33,117 +39,300 @@
+            if ( fromConfig == null ) {
+                // Only the web extension is available.
+                return fromApp;
+            } else {  
+                // Both are available: Answer the configuration override, with the
+                // web extension set as a delegate.
+                fromConfig.setDelegate(fromApp);
+                return fromConfig;
+            }
+
+        } else {
+            // Only the configuration override is available.
+            return fromConfig;
+        }
     }
-private com.ibm.ws.javaee.ddmodel.webext.WebExtComponentImpl getConfigOverrides(OverlayContainer overlay, ArtifactContainer artifactContainer) throws UnableToAdaptException {
-     if (configurations == null || configurations.isEmpty())
+
+    /**
+     * Answer the web extension type configuration override of a module.
+     *
+     * @param overlay The overlay container of the module.
+     * @param artifactContainer The raw container of the module.
+     *
+     * @return The web extension override of the module.
+     *
+     * @throws UnableToAdaptException Thrown if an error occured while
+     *     accessing the non-persistent cache.
+     */
+    private com.ibm.ws.javaee.ddmodel.webext.WebExtComponentImpl getConfigOverrides(
+        OverlayContainer overlay,
+        ArtifactContainer artifactContainer) throws UnableToAdaptException {
+
+        // No match is possible, and no errors are possible, when there are no
+        // web extension configuration overrides.
+
+        if ( (configurations == null) || configurations.isEmpty() ) {
           return null;
+        }
 
-     ApplicationInfo appInfo = (ApplicationInfo) overlay.getFromNonPersistentCache(artifactContainer.getPath(), ApplicationInfo.class);
-        ModuleInfo moduleInfo = null;
-if (appInfo == null) {
+        // Maybe application information is directly available ...
+        ApplicationInfo appInfo = (ApplicationInfo)
+            overlay.getFromNonPersistentCache(artifactContainer.getPath(), ApplicationInfo.class);
 
-          moduleInfo = (ModuleInfo) overlay.getFromNonPersistentCache(artifactContainer.getPath(), ModuleInfo.class);
-          if (moduleInfo == null)
+        // Or, maybe module information is available and application information can
+        // be obtained from that ...
+        ModuleInfo moduleInfo = null;
+        if ( appInfo == null ) {
+            moduleInfo = (ModuleInfo) overlay.getFromNonPersistentCache(artifactContainer.getPath(), ModuleInfo.class);
+            if ( moduleInfo == null ) {
                return null;
-          appInfo = moduleInfo.getApplicationInfo();
-     }
-     NestedConfigHelper configHelper = null;
-     if (appInfo != null && appInfo instanceof ExtendedApplicationInfo)
-          configHelper = ((ExtendedApplicationInfo) appInfo).getConfigHelper();
-      if (configHelper == null)
+            }
+            appInfo = moduleInfo.getApplicationInfo();
+            if ( appInfo == null ) {
```

Comment:
So this means if there is no application.xml we won't return a config override for the web module?

Reply:
That 'applicationInfo' is not the same as the application descriptor.  Later on, the application descriptor is retrieved (line 300 in the update code) with:

Application appDD = appInfo.getContainer().adapt(Application.class);

The (appInfo == null) check was previously next to the type check.  I moved the check to more clearly indicate that the goal was obtain application information.

Original lines 92-97:

          appInfo = moduleInfo.getApplicationInfo();
     }
     NestedConfigHelper configHelper = null;
     if (appInfo != null && appInfo instanceof ExtendedApplicationInfo)
          configHelper = ((ExtendedApplicationInfo) appInfo).getConfigHelper();
      if (configHelper == null)

There is a lot of particular code in this method which has to do with different cases of obtaining module information and application information, but I don't have any explanation other than what is in the code: Whether module information is available impacts both the override selection, and impacts whether the name matching validation is done.  But there is no explanation of when the module information would or would not be null.